src/game.js

```diff
@@ -0,0 +1,42 @@
+function Game (frameConstructor = Frame, finalFrameConstructor = FinalFrame) {
+  this._frames = []
+  this._finished = false
+  this.frameConstructor = frameConstructor
+  this.finalFrameConstructor = finalFrameConstructor
+}
+
+Game.prototype.getFrames = function () {
+  return this._frames
+}
+
+Game.prototype.getCurrentFrame = function () {
+  return this._frames[this._frames.length - 1]
+}
+
+Game.prototype.startNextFrame = function () {
+  if (this._frames.length === 9) {
+    this._frames.push(new this.finalFrameConstructor())
+  } else {
+    this._frames.push(new this.frameConstructor())
+  }
+}
+
+Game.prototype.getCurrentScore = function () {
+  return this._frames.reduce(function (total, frame, index, frames) {
+    return total + frame.getScore(frames[index + 1], frames[index + 2])
+  }, 0)
+}
+
+Game.prototype.addBowl = function (pins) {
+  if (this.isFinished()) { throw 'Cannot add bowl - game over!' }
+  if (this.getCurrentFrame().isFinished()) { this.startNextFrame() }
+  this.getCurrentFrame().addBowl(pins)
+  this.getCurrentScore()
+}
+
+Game.prototype.isFinished = function () {
+  if (this._frames.length >= 10 && this.getCurrentFrame().isFinished()) {
+    this._finished = true
+    return this._finished
```

Comment:
I don't know if this is intentional, but lines 39 and 40 are a great pattern. It's tempting to set something and then return the value we set, rather than looking up the value we just changed wherever we just changed it – but you didn't do that, and that's a good thing.

It's especially important when there might be a lag between setting something and doing the next step – and when you hit Promises (in week 7) and `async`/`await`, you'll see how important this is.